packages/ld-key-pair/KeyPair.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * defines the ability to sign based on a KeyPair or only
+ * from a private key
+ */
+interface Signer {
+    privateKey: string;
+    keyPair?: KeyPair;
+    sign(options: {
+        data: any;
+    }): string;
+}
+
+/**
+ * defines the ability to perform verification based on a KeyPair
+ * or a public key
+ */
+interface Verifier {
+    publicKey: string;
+    keyPair?: KeyPair;
+    verify(options: {
+        data: any;
+        signature: string;
+    }): boolean;
+}
+
+/**
+ * public key side of a key pair
+ */
+interface PublicNode {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    publicKeyBase58?: string;
+}
+/**
+ * public key side of a key pair
+ */
+ interface PrivateNode {
```

Comment:
I think this interface is really 2 things....

KeyPair (privateKey: Uint8Array), PrivateNode (privateKeyBase58 : string)

(assuming non binary linked data forms) You would never expose "privateKey: Uint8Array" in a linked data graph, because it has no compact string representation